File: week-2/problem-set/readability.c
#include "cs50.h"
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sen(string text);
int calculate_grade(int letters, int words, int sen);

int main(void) {
  string text = get_string("Text: ");
  // Calculating grade by finding number of letters, words and sentences
  int grade =
      calculate_grade(count_letters(text), count_words(text), count_sen(text));
  // Printing grade according to given ranges
  if (grade < 1) {
    printf("Before Grade 1");
  } else if (grade >= 16) {
    printf("Grade 16+");
  } else {
    printf("Grade %i", grade);
  }
  printf("\n");
}

int count_letters(string text) {
  int letters = 0;
  for (int i = 0; i < strlen(text); i++) {
    // Incrementing the sum if the character of the string is an alphabet
    if (isalpha(text[i])) {
      letters++;
    }
  }
  return letters;
}

int count_words(string text) {
  // Setting words to 1 as for n words, there are (n-1) whitespaces in between
  int words = 1;
  for (int i = 0; i < strlen(text); i++) {
    if (text[i] == ' ') {
      words++;
    }
  }
  return words;
}

int count_sen(string text) {
  int sentences = 0;
  for (int i = 0; i < strlen(text); i++) {
    char tempchar = text[i];
    // Checking if a string has begun by validating the end of line character
    if (tempchar == '?' || tempchar == '.' || tempchar == '!') {
      sentences++;
    }
  }
  return sentences;
}

int calculate_grade(int letters, int words, int sen) {
  // Calculating average number of letters for 100 words
  float L = (float)letters / words * 100;
  // Calculating average number of sentences for 100 words
  float S = (float)sen / words * 100;
  // Computing the Coleman-Liau index
  int index = round(0.0588 * L - 0.296 * S - 15.8);
  return index;
}

File: week-6/problem-set/readability.py
# TODO


def main():
    text = input("Text: ")
    grade = calculate_grade(count_letters(text), count_words(text), count_sen(text))
    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print("Grade ", grade)
    print("\n")


def count_letters(text):
    letters = 0
    i = 0
    while i < len(text):
        if text[i].isalpha():
            letters += 1
        i += 1
    return letters


def count_words(text):
    words = 1
    i = 0
    while i < len(text):
        if text[i] == " ":
            words += 1
        i += 1
    return words


def count_sen(text):
    sentences = 0
    i = 0
    while i < len(text):
        tempchar = text[i]
        if tempchar == "?" or tempchar == "." or tempchar == "!":
            sentences += 1
        i += 1
    return sentences


def calculate_grade(letters, words, sen):
    L = letters / words * 100
    S = sen / words * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)
    return index


main()
